core/embed/rust/src/ui/display/font.rs

```diff
@@ -198,31 +271,27 @@ impl Font {
             return 0;
         }
 
-        if let Some(glyph) = self.get_first_glyph_from_text(text) {
-            glyph.bearing_x
-        } else {
-            0
-        }
+        text.chars().next().map_or(0, |c| {
+            self.with_glyph_data(|data| data.get_glyph(c).bearing_x)
+        })
     }
 
     pub fn char_width(self, ch: char) -> i16 {
-        display::char_width(ch, self.into())
+        self.with_glyph_data(|data| data.get_glyph(ch).adv)
```

Comment:
I just removed the delegation to `char_width` and instead put the whole loop in `with_glyph_data` and get the `advance` from glyph data where needed. I don't think it's too much repetition..
77b3d79e01625e077d88fcf01125ba75a559cd25

Good catch, I initially did the same mistake with other functions... I wrote `for ... with_data(|..| )` instead of `with_data(|..| for ...)`

Reply:
Made a mistake 74ad10d44127b41b970a1d7e643abe64f2bcfd11